docker-compose.yml

```diff
@@ -1,15 +1,111 @@
-version: "3.8"
+version: '3.9'
+
+networks:
+  trainingdairy:
+    driver: bridge
 
 services:
+  nginx:
+    image: nginx:stable-alpine
+    tty: true
+    ports:
+      - "3001:80"
+    volumes:
+      - ./backend:/var/www/html
+      - ./bin/.nginx/default.conf:/etc/nginx/conf.d/default.conf
+    depends_on:
+      - php
+      - mysql
+    networks:
+      - trainingdairy
 
-  backend:
+  php:
     build:
       context: .
-      target: base
+      dockerfile: Dockerfile
+    tty: true
     environment:
-      - NODE_ENV=development
+      SERVICE_NAME: php
+      SERVICE_TAGS: dev
+      DB_HOST: ${DB_HOST}
+      DB_NAME: ${DB_NAME}
+      DB_USER: ${DB_USER}
+      DB_PASSWORD: ${DB_PASSWORD}
+      XDEBUG_MODE: develop,debug
+      XDEBUG_CONFIG:
+        client_host=host.docker.internal
+        start_with_request=yes
     volumes:
-      - ./backend:/var/www/html:rw
-      - ./frontend/dist:/var/www/html/dist/:rw
+      - ./backend:/var/www/html
+      - ./bin/.php/local.ini:/usr/local/etc/php/conf.d/local.ini
+      - ./bin/.php/conf.d/xdebug.ini:/usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini
+      - ./bin/.php/conf.d/conf.d/error_reporting.ini:/usr/local/etc/php/conf.d/error_reporting.ini
+    ports:
+      - "80:9000"
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+    networks:
+      - trainingdairy
+
+  mysql:
+    image: mysql:5.7.29
+    platform: linux/x86_64
+    command: mysqld --sql_mode=""
+    restart: always
     ports:
-      - 3001:80
\ No newline at end of file
+      - "3306:3306"
+    environment:
+      MYSQL_DATABASE: laravel_app
+      MYSQL_PASSWORD: secret
+      MYSQL_ROOT_PASSWORD: secret456
+      SERVICE_TAGS: dev
+      SERVICE_NAME: mysql
+    networks:
+      - trainingdairy
+    volumes:
+      - ./bin/.sql_scripts:/docker-entrypoint-initdb.d
+      - ./bin/.mysql:/var/lib/mysql
+
+  composer:
+    image: composer:latest
+    user: "1000:1000"
+    container_name: backend_composer
+    volumes:
+      - ./backend:/var/www/html
+    working_dir: /var/www/html
+    command: composer update
+    depends_on:
+      - php
+    networks:
+      - trainingdairy
```

Comment:
@AnimaVillis we now have a separate docker service for composer. We can install new packages in two ways. I'll provide examples below.

1. Changing temporarily the `command` inside the docker-compose (**do not commit this change**)
2. Adding new package name in the `composer.json`

## Examples
Let's say we don't have a package called `bcosca/fatfree-core` and I want to install it.

### 1 - Modify command field
1. Find `command` key in the composer service in `docker-compose.yml`, currently at line **76**
2. Change it's value from `composer update` to (whatever command you need) `composer install bcosca/fatfree-core`
3. Apply changes by running the container from the terminal `docker-compose up composer`
4. Done

### 2 - Inline in json

1. Add to `composer.json` a new line
```json
"bcosca/fatfree-core": "^3.8"
```
2. Always choose the latest version. If you don't know what is the latest make sure to install suggested extension from the `trainingdiary.code-workspace` file and VSCode will suggest you the latest as soon as you add `""` after the package name 😁 
3. Apply changes by running the composer service `docker-compose up composer`
4. Done


```